### ISSUE TITLE
Wrong solution result for non square matrix.

### ISSUE THREAD

**Issue by Gluttton** [saebyn/munkres-cpp#10]
Based on test: ```MunkresTest.solve_3x2_NonObviousSolutionCase001_Success``` solution result for non square matrix is wrong.

Possibly reason (in my opinion):
1. Wrong test case.
2. Algorithm designed only for square matrix (so special assertion is required).
3. Error in algorithm.

**Comment by Gluttton** [comment]
The problem have not solved completely.

[Here is](https://github.com/saebyn/munkres-cpp/issues/20) the related issue.
Also commits
0acab86c220b839f5dec6ff68e66e4ef2ab10707
5b66cd73a5b85bba5544ebe3bd852c29fc434fa7
were added and related to the issue.

The problem is required more deeper research.

**Comment by Gluttton** [comment]
I've simplified test data with preserving their behaviour.
Instead of:
```
  Matrix<double> etalon_matrix{
    {-1.0, -1.0, -1.0, -1.0},
    { 0.0, -1.0, -1.0, -1.0},
    {-1.0,  0.0, -1.0, -1.0},
    {-1.0, -1.0, -1.0, -1.0},
    {-1.0, -1.0, -1.0,  0.0},
    {-1.0, -1.0,  0.0, -1.0}
  };
  Matrix<double> test_matrix{
    {1.79769e+308, 7.33184e+08,  9.41561e+08,  2.79247e+08},
    {3.06449e+08,  1.79769e+308, 3.3464e+08,   7.06878e+08},
    {9.93296e+08,  1.9414e+08,   1.79769e+308, 1.14174e+08},
    {3.51623e+08,  2.48635e+08,  7.81242e+08,  1.79769e+308},
    {7.02639e+08,  8.51663e+08,  9.37382e+08,  4.96945e+07},
    {7.58851e+08,  8.58445e+08,  8.7235e+07,   5.47076e+08}
  };
```
I use:
```
  Matrix<double> etalon_matrix{
    {-1.0, -1.0},
    { 0.0, -1.0},
    {-1.0,  0.0}
  };
  Matrix<double> test_matrix{
    {1.0e+17, 3},
    {2,       1.0e+17},
    {4,       1}
  };
```
etc...

After playing with simple tests I found out that my test cases for 7x5 and 5x7 matrices was wrong.

So it looks like the problem was fixed with commit 0acab86c220b839f5dec6ff68e66e4ef2ab10707 .
